bin/plugin/commands/release.js

```diff
@@ -0,0 +1,817 @@
+ * @param {string}  version       Released version.
+ * @param {string}  versionLabel  Label of the released Version.
+ * @param {string}  changelog     Release changelog.
+ * @param {boolean} isPrerelease  is a pre-release.
+ * @param {string}  abortMessage  Abort message.
+ * @return {Object} Github release object.
+ */
+async function runGithubReleaseStep(
+	zipPath,
+	version,
+	versionLabel,
+	changelog,
+	isPrerelease,
+	abortMessage
+) {
+	let octokit;
+	let release;
+	await runStep( 'Creating the GitHub release', abortMessage, async () => {
+		await askForConfirmation(
+			'Proceed with the creation of the GitHub release?',
+			true,
+			abortMessage
+		);
+
+		const { token } = await inquirer.prompt( [
+			{
+				type: 'input',
+				name: 'token',
+				message:
+					'Please provide a GitHub personal authentication token. Navigate to ' +
+					formats.success(
+						'https://github.com/settings/tokens/new?scopes=repo,admin:org,write:packages'
+					) +
+					' to create one.',
+			},
+		] );
+
+		octokit = new Octokit( {
+			auth: token,
+		} );
+
+		const releaseData = await octokit.repos.createRelease( {
+			owner: config.githubRepositoryOwner,
+			repo: config.githubRepositoryName,
+			tag_name: 'v' + version,
+			name: versionLabel,
+			body: changelog,
+			prerelease: isPrerelease,
+		} );
+		release = releaseData.data;
+
+		log( '>> The GitHub release has been created.' );
+	} );
+	abortMessage =
+		abortMessage + ' Make sure to remove the the GitHub release as well.';
+
+	// Uploading the Zip to the Github release
+	await runStep( 'Uploading the plugin ZIP', abortMessage, async () => {
+		const filestats = fs.statSync( zipPath );
+		await octokit.repos.uploadReleaseAsset( {
+			url: release.upload_url,
+			headers: {
+				'content-length': filestats.size,
+				'content-type': 'application/zip',
+			},
+			name: config.slug + '.zip',
+			file: fs.createReadStream( zipPath ),
+		} );
+		log( '>> The plugin ZIP has been successfully uploaded.' );
+	} );
+
+	log(
+		'>> The GitHub release is available here: ' +
+			formats.success( release.html_url )
+	);
+
+	return release;
+}
+
+/**
+ * Updates and commits the content of the SVN repo using the new plugin ZIP.
+ *
+ * @param {string} svnWorkingDirectoryPath SVN Working Directory Path.
+ * @param {string} zipPath                 Plugin zip path.
+ * @param {string} version                 Version.
+ * @param {string} abortMessage            Abort Message.
+ */
+async function runUpdateTrunkContentStep(
+	svnWorkingDirectoryPath,
+	zipPath,
+	version,
+	abortMessage
+) {
+	// Updating the content of the svn
+	await runStep( 'Updating trunk content', abortMessage, async () => {
+		log( '>> Replacing trunk content using the new plugin ZIP' );
+
+		const readmePath = svnWorkingDirectoryPath + '/readme.txt';
+
+		const previousReadmeFileContent = fs.readFileSync( readmePath, 'utf8' );
+		const stableTag = previousReadmeFileContent.match(
+			STABLE_TAG_REGEX
+		)[ 0 ];
```

Comment:
Not sure how much we should trust the match to exist:

```
bin/plugin/commands/release.js:521:21 - error TS2531: Object is possibly 'null'.

521  	const stableTag = previousReadmeFileContent.match(
     	                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
522  		STABLE_TAG_REGEX
    ~~~~~~~~~~~~~~~~~~~
523  	)[ 0 ];
```

Reply:
Well, if it doesn't exist, I believe the file is corrupted somehow.